File: backend/src/models/work.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Work {
    pub id: String,
    pub title: String,
    pub major_code: String,
    pub major_name: String,
    pub theme: String,
    #[sqlx(json)]
    pub tags: Vec<String>,
    #[sqlx(json)]
    pub authors: Vec<String>,
    pub teacher: String,
    pub poster_url: Option<String>,
    pub video_url: Option<String>,
    pub duration: Option<String>,
    pub introduction: Option<String>,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub updated_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Deserialize)]
pub struct ListWorksQuery {
    pub major: Option<String>,
    pub page: Option<i32>,
    pub page_size: Option<i32>,
}

#[derive(Debug, Serialize)]
pub struct WorkListResponse {
    pub items: Vec<WorkWithStats>,
    pub page: i32,
    pub page_size: i32,
    pub total: i64,
    pub total_pages: i32,
}

#[derive(Debug, Serialize)]
pub struct WorkWithStats {
    #[serde(flatten)]
    pub work: Work,
    pub like_count: i32,
    pub comment_count: i32,
}

File: frontend/types/work.ts
// 作品数据类型定义（与后端 API 返回格式匹配）
export interface Work {
  id: string
  title: string
  major_code: string
  major_name: string
  theme: string
  tags: string[]
  authors: string[]
  teacher: string
  poster_url: string
  video_url: string
  duration: string
  introduction: string
  like_count: number
  comment_count: number
  created_at: string
  updated_at: string
}

export interface WorkListResponse {
  items: Work[]
  page: number
  page_size: number
  total: number
  total_pages: number
}
